From 7ca1d8554eb0ffb267f91a7810fcf0ebdca4980c Mon Sep 17 00:00:00 2001
From: Shen Zhenyi <szy@rock-chips.com>
Date: Mon, 14 Dec 2015 09:47:01 +0800
Subject: [PATCH] rk_fb: screen info add screen type and interlace

When box is starting, CVBS display then HDMI accessed
and pulled out in kernel stage, FB keeps HDMI state but
display interface is CVBS, android need know that.

Change-Id: Ie2b75fa72f3201383c97f702976d4f0cd95f00e2
Signed-off-by: Shen Zhenyi <szy@rock-chips.com>
---
 drivers/video/rockchip/rkfb_sysfs.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/rkfb_sysfs.c b/drivers/video/rockchip/rkfb_sysfs.c
index 1bf795e08b3d..b48af77351e8 100644
--- a/drivers/video/rockchip/rkfb_sysfs.c
+++ b/drivers/video/rockchip/rkfb_sysfs.c
@@ -104,8 +104,10 @@ static ssize_t show_screen_info(struct device *dev,
 
 	if (ft > 0)
 		fps = div64_u64(1000000000000llu, ft);
-	return snprintf(buf, PAGE_SIZE, "xres:%d\nyres:%d\nfps:%d\n",
-			screen->mode.xres, screen->mode.yres, fps);
+	return snprintf(buf, PAGE_SIZE,
+			"xres:%d\nyres:%d\nfps:%d\ntype:%d\ninterlace:%d\n",
+			screen->mode.xres, screen->mode.yres,
+			fps, screen->type, screen->mode.vmode);
 }
 
 static ssize_t show_disp_info(struct device *dev,
-- 
2.35.3

